src/serialize.h

```diff
@@ -342,6 +342,12 @@ class CFlatData
         pbegin = (char*)begin_ptr(v);
         pend = (char*)end_ptr(v);
     }
+    template <unsigned int N, typename T, typename S, typename D>
```

Comment:
when I name template parameters, I usually use more comprehensible names so it's easy to understand what's going on here. But probably it's a part of current implementation, and it's hard to name them differently.